src/rust/catmem/mod.rs

```diff
@@ -335,8 +347,15 @@ impl CatmemLibOS {
 
     /// Takes out the [OperationResult] associated with the target [SchedulerHandle].
     fn take_result(&mut self, handle: TaskHandle) -> (QDesc, OperationResult) {
-        let task: OperationTask = OperationTask::from(self.scheduler.remove(handle).as_any());
-        task.get_result().expect("The coroutine has not finished")
+        let task: OperationTask = OperationTask::from(self.scheduler.remove(handle.clone()).as_any());
+        let (qd, result): (QDesc, OperationResult) = task.get_result().expect("The coroutine has not finished");
+
+        match self.qtable.borrow_mut().get_mut(&qd) {
+            Some(queue) => queue.remove_pending_op(&handle),
+            None => debug!("take_result(): this queue was closed (qd={:?})", qd),
+        }
+
+        (qd, result)
```

Comment:
Here you will also want to add a `handle.take_task_id()` to avoid issue #765

Reply:
We probably should but it won't be needed after PR #740 so I just left the issue